File: component-acgschedule/src/main/java/com/rabtman/acgschedule/mvp/contract/ScheduleMainContract.java
package com.rabtman.acgschedule.mvp.contract;

import com.rabtman.acgschedule.mvp.model.jsoup.DilidiliInfo;
import com.rabtman.common.base.mvp.IModel;
import com.rabtman.common.base.mvp.IView;
import io.reactivex.Flowable;

/**
 * @author Rabtman
 */

public interface ScheduleMainContract {

  interface View extends IView {

    void showDilidiliInfo(DilidiliInfo dilidiliInfo);

    void start2ScheduleVideo(String videoUrl);
  }

  interface Model extends IModel {

    Flowable<DilidiliInfo> getDilidiliInfo();
  }
}

File: component-acgschedule/src/main/java/com/rabtman/acgschedule/mvp/contract/ScheduleMainContract.kt
package com.rabtman.acgschedule.mvp.contract

import com.rabtman.acgschedule.mvp.model.jsoup.ScheduleInfo
import com.rabtman.common.base.mvp.IModel
import com.rabtman.common.base.mvp.IView
import io.reactivex.Flowable

/**
 * @author Rabtman
 */
interface ScheduleMainContract {
    interface View : IView {
        fun showScheduleInfo(scheduleInfo: ScheduleInfo)
        fun start2ScheduleVideo(videoUrl: String)
    }

    interface Model : IModel {
        fun getScheduleInfo(): Flowable<ScheduleInfo>
    }
}